ssl-config/src/main/java/com/palantir/ssl/KeyStoreConfiguration.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.ssl;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Style.ImplementationVisibility;
+
+@JsonDeserialize(as = ImmutableKeyStoreConfiguration.class)
+@JsonSerialize(as = ImmutableKeyStoreConfiguration.class)
+@Value.Immutable
+@Value.Style(visibility = ImplementationVisibility.PACKAGE)
+public abstract class KeyStoreConfiguration {
```

Comment:
Yep

From: nmiyake <notifications@github.com<mailto:notifications@github.com>>
Reply-To: palantir/http-remoting <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, February 9, 2016 at 3:08 PM
To: palantir/http-remoting <http-remoting@noreply.github.com<mailto:http-remoting@noreply.github.com>>
Cc: Mark Elliot <melliot@palantir.com<mailto:melliot@palantir.com>>
Subject: Re: [http-remoting] Add ability to specify keystore for SSLSocketFactory (#35)


In ssl-config/src/main/java/com/palantir/ssl/KeyStoreConfiguration.java<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_http-2Dremoting_pull_35-23discussion-5Fr52390164&d=CwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=63CHyb-LL-0G2TTkGbAjWE8fVaE92hUVMw6sMOvJvgs&s=QeLweIYz664NbPdTrfX0MLo2DR0IATB2rtJ8vwvq1Cc&e=>:

> +
> +import static com.google.common.base.Preconditions.checkArgument;
> +
> +import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
> +import com.fasterxml.jackson.databind.annotation.JsonSerialize;
> +import com.google.common.base.Optional;
> +import java.nio.file.Path;
> +import java.nio.file.Paths;
> +import org.immutables.value.Value;
> +import org.immutables.value.Value.Style.ImplementationVisibility;
> +
> +@JsonDeserialize(as = ImmutableKeyStoreConfiguration.class)
> +@JsonSerialize(as = ImmutableKeyStoreConfiguration.class)
> +@Value.Immutable<mailto:+@Value.Immutable>
> +@Value.Style<mailto:+@Value.Style>(visibility = ImplementationVisibility.PACKAGE)
> +public abstract class KeyStoreConfiguration {


Idea is that, if specified, alias will be used to choose the specific key for keystore? I'm fine with that -- provides more fine-grain control and doesn't impact the main use case since it's optional.

—
Reply to this email directly or view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_http-2Dremoting_pull_35_files-23r52390164&d=CwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=63CHyb-LL-0G2TTkGbAjWE8fVaE92hUVMw6sMOvJvgs&s=VU7L3qDeuF7pskZqgX5fQ6EhsdAETAW_2OAagCqhozw&e=>.